modules/compiler/test/lit/passes/reqd_work_group_size_operands.ll

```diff
@@ -0,0 +1,54 @@
+; Copyright (C) Codeplay Software Limited
+;
+; Licensed under the Apache License, Version 2.0 (the "License") with LLVM
+; Exceptions; you may not use this file except in compliance with the License.
+; You may obtain a copy of the License at
+;
+;     https://github.com/codeplaysoftware/oneapi-construction-kit/blob/main/LICENSE.txt
+;
+; Unless required by applicable law or agreed to in writing, software
+; distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+; WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+; License for the specific language governing permissions and limitations
+; under the License.
+;
+; SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+; RUN: muxc --passes work-item-loops,verify -S %s | FileCheck %s --check-prefix=CHECK1
```

Comment:
I'm not sure how reliable the order of the output function is, will it always stay the same when changing LLVM version etc?

Reply:
Ah I see what you're doing. I think it's fair to assume that the order is stable. We do it almost everywhere else in our LIT tests.

Though I was wondering if a `UnitCompiler` unit test might be preferable? Then we don't need to assume about the IR (neither the order of functions nor what comparisons we check against).

Reply:
I suggested a unit test because it's by far the best way of testing the API and the change to it. There's no guarantee that the work-item-loops pass actually parses that metadata, and if it does it might not use that API, and there's no guarantee it emits those comparisons, nor in that order. It's far flakier with a LIT test.